src/test/java/ch/unisg/ics/interactions/wot/td/io/IDGraphWriterTest.java

```diff
@@ -0,0 +1,185 @@
+      .setContentType("application/json")
+      .setMethodName("POST")
+      .build();
+
+    this.GET_FORM = new Form.Builder("http://example.org/action")
+      .addOperationType(TD.readProperty)
+      .setMethodName("GET")
+      .build();
+
+    this.STRING_SCHEMA  = new StringSchema.Builder().build();
+  }
+
+  @Test
+  public void testWriteCompleteID() throws IOException {
+    InteractionDescription completeIntD = InteractionDescription.builder()
+      .title(ID_TITLE)
+      .input(new InteractionInput("input1", POST_FORM, STRING_SCHEMA))
+      .output(new InteractionOutput("output1", STRING_SCHEMA))
+      .type(InteractionTypes.ACTION)
+      .uri("http://example.org/log#interaction-1")
+      .build();
+
+    String testID = PREFIXES + String.format(
+      "\n<http://example.org/log#interaction-1> a log:ActionLog;\n" +
+      "  td:title \"interaction-1\";\n" +
+      "  td:hasBase <http://example.org/log#interaction-1>;\n" +
+      "  log:created \"%s\";\n" +
+      "  td:hasInput [ a <https://www.w3.org/2019/wot/json-schema#StringSchema>;\n" +
+      "      td:value \"input1\";\n" +
+      "      td:hasForm [\n" +
+      "          htv:methodName \"POST\";\n" +
+      "          hctl:hasTarget <http://example.org/action>;\n" +
+      "          hctl:forContentType \"application/json\";\n" +
+      "          hctl:hasOperationType td:invokeAction\n" +
+      "        ]\n" +
+      "    ];\n" +
+      "  td:hasOutput [ a <https://www.w3.org/2019/wot/json-schema#StringSchema>;\n" +
+      "      td:value \"output1\"\n" +
+      "    ] .\n", getDateTimeWithoutSeconds());
+
+    assertIsomorphicGraphs(testID, completeIntD);
+  }
+
+  @Test
+  public void testWriteIDWithoutInput() throws IOException {
+    InteractionDescription intDWithoutInput = InteractionDescription.builder()
+      .title(ID_TITLE)
+      .type(InteractionTypes.EVENT)
+      .output(new InteractionOutput("output1", STRING_SCHEMA))
+      .uri("http://example.org/log#interaction-1")
+      .build();
+
+    String testID = PREFIXES + String.format(
+      "\n<http://example.org/log#interaction-1> a log:EventLog;\n" +
+        "  td:title \"interaction-1\";\n" +
+        "  td:hasBase <http://example.org/log#interaction-1>;\n" +
+        "  log:created \"%s\";\n" +
+        "  td:hasOutput [ a <https://www.w3.org/2019/wot/json-schema#StringSchema>;\n" +
+        "      td:value \"output1\"\n" +
+        "    ] .\n", getDateTimeWithoutSeconds());
+
+    assertIsomorphicGraphs(testID, intDWithoutInput);
+  }
+
+  @Test
+  public void testWriteIDWithoutInputValue() throws IOException {
+    InteractionDescription requestIntD = InteractionDescription.builder()
+      .title(ID_TITLE)
+      .input(new InteractionInput(GET_FORM))
+      .output(new InteractionOutput("output1", STRING_SCHEMA))
+      .type(InteractionTypes.ACTION)
+      .uri("http://example.org/log#interaction-1")
+      .build();
+
+    String testID = PREFIXES + String.format(
```

Comment:
Thank you very much, @flooji. Please, do not update the implementation based on the following. They are up to discussion:
- `td:title` and `td:hasBase` which are defined in the WoT TD ontology have a domain of `td:Thing`. Therefore, they cannot be used here unless `log:Action` is a subclass of `td:Thing`. This is not the case here so I propose to use `log:hasTitle` and `log:hasBase` instead if such properties are needed. 
- I would argue though that neither properties are required. `td:hasBase` is being used in TDs to simplify the description of the targets in an `hctl:Form`. Here, the interaction resource does not describe its `td:InteractionAffordances` so I would say that this is not required. I also don't have a use case for using `log:hasTitle`.
- `td:hasOutput` and `td:hasInput` are not part of the WoT TD Ontology. I would suggest `log:hasOutput` and `log:hasInput`. 
- `td:value` is not part of the WoT TD Ontology. Also, note that a an Input is different than its schema (an input schema). I would propose the creation of a `log:Input` and a `log:Output`. Both have a `log:value` and a `log:hasSchema` (whose range can be a `js:DataSchema`.

Reply:
Hi Danai, Thanks for your input. I agree with your points, but wait with the implementation as suggested.